src/components/posts-index.js

```diff
@@ -0,0 +1,100 @@
+
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchWeather, setDefault } from '../actions';
+import { useForm } from "react-hook-form";
+import _ from 'lodash';
+import { useEffect } from 'react';
+import { Sparklines, SparklinesReferenceLine, SparklinesLine } from 'react-sparklines';
+
+const WeatherIndex = () => {
+  const { register, handleSubmit } = useForm();
+  const posts = useSelector(state => state.posts);
+  const dispatch = useDispatch();
+
+  //checks if default is set and if so fetch default weather
+  useEffect(() => {
+    if (sessionStorage.getItem('default') !== null) {
+      dispatch(fetchWeather(sessionStorage.getItem('default')))
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [fetchWeather]);
+
+  //on search click handler, fetches weather
+  const onSubmit = (data) => {
+    dispatch(fetchWeather(data.cityName))
+  };
+
+  //set default click handler
+  const handleSetDefault = (data, e) => {
+    e.target.className = "btn btn-outline-secondary btn-sm"
+    e.target.disabled = true;
+    e.target.firstChild.data = 'Default set!'
```

Comment:
It's usually best to avoid manipulating the DOM directly like this when using React. There is a more "React" way to do this.